src/main/java/com/minecolonies/coremod/colony/VisitorData.java

```diff
@@ -114,10 +115,14 @@ public void updateEntityIfNecessary()
     {
         if (getEntity().isPresent())
         {
-            return;
+            final Entity entity = getEntity().get();
+            if (entity.isAlive() && entity.addedToChunk && WorldUtil.isEntityBlockLoaded(entity.world, entity.getPosition()))
+            {
+                return;
+            }
         }
 
-        if (WorldUtil.isEntityBlockLoaded(getColony().getWorld(), getLastPosition()))
+        if (getLastPosition() != BlockPos.ZERO && WorldUtil.isEntityBlockLoaded(getColony().getWorld(), getLastPosition()))
         {
             getColony().getVisitorManager().spawnOrCreateCivilian(this, getColony().getWorld(), getLastPosition(), true);
```

Comment:
I'm pretty sure this causes the out of world issues

Reply:
ye might do